kong/pdk/vault.lua

```diff
@@ -168,6 +176,71 @@ local function new(self)
   end
 
 
+  local function get_prefix_hash(schema, prefix, base_config)
+    local hash = CONFIG_HASHES[prefix]
+    if hash then
+      return hash
+    end
+
+    for k in schema:each_field() do
+      local v = base_config[k]
+      if v ~= nil then
+        if not hash then
+          hash = true
+          KEY_BUFFER:reset()
+        end
+        KEY_BUFFER:putf("%s=%s;", k, v)
+      end
+    end
+
+    if hash then
+      hash = encode_base64url(md5_bin(KEY_BUFFER:get()))
+      CONFIG_HASHES[prefix] = hash
+    end
+
+    return hash
+  end
+
+
+  local function get_config(schema, base_config, config_overrides, prefix)
+    if not config_overrides or isempty(config_overrides) then
+      if not prefix then
+        return base_config
+      end
+
+      return base_config, get_prefix_hash(schema, prefix, base_config)
+    end
+
+    local config
+    for k, f in schema:each_field() do
+      local v = config_overrides[k]
+      v = arguments.infer_value(v, f)
+      -- TODO: should we be more visible with validation errors?
+      if v ~= nil and schema:validate_field(f, v) then
```

Comment:
First we need to think what we should do about these validation errors. As it can happen very often, e.g. if there is a typo in vault reference, I don't want it to flood the logs. All we can pretty much do is to ignore them and perhaps log. We may need to add validation to schemas, so that we do deeper validation of reference urls, but currently we just check the format.

Reply:
I think it's better to "flood" logs with something you have to pay attention to rather than keeping silent.